lib/Sema/TypeChecker.cpp

```diff
@@ -714,13 +714,14 @@ void swift::performTypeChecking(SourceFile &SF, TopLevelContext &TLC,
 
     typeCheckFunctionsAndExternalDecls(TC);
   }
-  MyTC.reset();
 
   // Checking that benefits from having the whole module available.
   if (!(Options & TypeCheckingFlags::DelayWholeModuleChecking)) {
     performWholeModuleTypeChecking(SF);
   }
 
+  MyTC.reset();
```

Comment:
I don't think it was for efficiency, just to make sure that we didn't try to do any checks that depended on having a type resolver. We've added more uses of a resolver (type checker) since then, so it may no longer be valid.